src/app/pages/server-details/server-details.component.ts

```diff
@@ -1159,4 +1167,41 @@ export class ServerDetailsComponent implements OnInit {
       return this.serverDetails.benchmark_scores?.find((b) => b.benchmark_id === 'stress_ng:cpu_all' && (b.config as any)?.cores !== 1)?.score?.toFixed(0) || '-';
     }
   }
+
+  generateSchemaJSON(title: string, description: string, keywords: string) {
+    if(!this.serverDetails) {
+      return;
+    }
+
+    console.log('Generating JSON-LD schema for server details page', this.serverDetails);
+
+    const json: any =
+    // generate schema.org Product JSON-LD
+    //const server = this.serverDetails;
+    {
+      "@context": "https://schema.org/",
+      "@type": "Product",
+      "name": this.serverDetails.display_name,
+      "description": description,
+      "brand": {
+        "@type": "Organization",
+        "name": this.serverDetails.vendor.name,
+        "logo": this.serverDetails.vendor.logo,
+        "url": this.serverDetails.vendor.homepage
+      }
+    }
+
+    if(this.similarByFamily.length) {
+      json['isSimilarTo'] = this.similarByPerformance.map((s) => {
+        return {
+          "@type": "Product",
+          "name": s.display_name,
+          "url": `https://www.sparecores.com/server/${s.vendor_id}/${s.api_reference}`
+        }
+      });
+    }
+
```

Comment:
it seems that we need to provide pricing as well:

![image](https://github.com/SpareCores/sc-www/assets/495736/ba085206-e072-4cd3-ae5f-257c4a25f547)

I think offers/price is what we can do (including the lowest) as per https://developers.google.com/search/docs/appearance/structured-data/product-snippet#structured-data-type-definitions